File: Practice/Python/Algorithms/Searching-Sorting-Algorithms/SelectionSort.py
# Selection sort in Python
# A Selection sort is a sorting algorithm that is used to sort items of a list.
def selectionSort(array, size): # function to swap elements at the given index values
    for step in range(size):
        min_idx = step
        for i in range(step + 1, size):
            # to sort in descending order, change > to < in this line select the minimum element in each loop
            if array[i] < array[min_idx]:
                min_idx = i
        # put min at the correct position
        (array[step], array[min_idx]) = (array[min_idx], array[step])
data = [-2, 45, 0, 11, -9]
size = len(data)
selectionSort(data, size)
print('Sorted Array in Ascending Order:')
print(data)

File: Practice/JavaScript/DSA/Algorithms/Searching-Sorting-Algorithms/SelectionSort.js
function swap(arr, xp, yp) {
  var temp = arr[xp];
  arr[xp] = arr[yp];
  arr[yp] = temp;
}

function selectionSort(arr, n) {
  var i, j, min_idx;

  for (i = 0; i < n - 1; i++) {
    min_idx = i;
    for (j = i + 1; j < n; j++) {
      if (arr[j] < arr[min_idx]) min_idx = j;
    }
    swap(arr, min_idx, i);
  }
}

function printArray(arr, size) {
  var i;
  for (i = 0; i < size; i++) console.log(arr[i] + " ");
  console.log();
}

var arr = [64, 25, 12, 22, 11];
var n = arr.length;
selectionSort(arr, n);
console.log("Sorted array: ");
printArray(arr, n); // 11 12 22 25 64
